src/cmake/thirdparty/SetupGeosxThirdParty.cmake

```diff
@@ -108,6 +108,7 @@ endif()
 ################################
 # HDF5
 ################################
+set(HDF5_DIR ${GEOSX_TPL_DIR}/hdf5)
```

Comment:
According to the [documentation](https://cmake.org/cmake/help/v3.0/command/if.html)
```
if(DEFINED <variable>)
True if the given variable is defined. It does not matter if the variable is true or false just if it has been set. (Note macro arguments are not variables.)
```
Which means that the following `if(DEFINED HDF5_DIR)` is made useless by your statement.
This is probably not what we want here (neither in the other statement below).

In `host-configs/tpls.cmake` we can see that
```
if(EXISTS ${GEOSX_TPL_DIR}/hdf5)
  set(HDF5_DIR ${GEOSX_TPL_DIR}/hdf5 CACHE PATH "" FORCE)
endif()
```
so `HDF5_DIR` should be defined there. Do you have the `GEOSX_TPL_DIR` defined? This is probably the place where you should check!

Reply:
Yes, I think this is the proper fix 😉